lnwallet/chanfunding/psbt_assembler.go

```diff
@@ -178,23 +177,26 @@ func (i *PsbtIntent) FundingParams() (btcutil.Address, int64, *psbt.Packet,
 	// The funding output needs to be known already at this point, which
 	// means we need to have the local and remote multisig keys bound
 	// already.
-	witnessScript, out, err := i.FundingOutput()
+	_, out, err := i.FundingOutput()
 	if err != nil {
 		return nil, 0, nil, fmt.Errorf("unable to create funding "+
 			"output: %v", err)
 	}
-	witnessScriptHash := sha256.Sum256(witnessScript)
 
-	// Encode the address in the human readable bech32 format.
-	addr, err := btcutil.NewAddressWitnessScriptHash(
-		witnessScriptHash[:], i.netParams,
-	)
+	script, err := txscript.ParsePkScript(out.PkScript)
+	if err != nil {
+		return nil, 0, nil, fmt.Errorf("unable to parse funding "+
+			"output script: %w", err)
+	}
+
+	// Encode the address in the human-readable bech32 format.
+	addr, err := script.Address(i.netParams)
```

Comment:
Unrelated to this PR, was curious about the implementation of `Address` and found it [here](https://github.com/btcsuite/btcd/blame/ec401d00a149d817e227b6afbc35b2399aca489d/txscript/pkscript.go#L152-L159). I notice we only use `addrs[0]` returned from `ExtractPkScriptAddrs`, and my question is, since multisig gives 2 address, is it ok to use `addr[0]` here?

Reply:
I forgot to actually use the custom channel type. Added the validation of the multisig output as well in the itest now. Can you take another look please?

Reply:
> nd my question is, since multisig gives 2 address, is it ok to use addr[0] here?

So this is a pretty old quirk: back when you could use bare multi-sig, you could actually look at a `pkScript` and directly extra the number of keys. But as of now, it doesn't ever do anything: https://github.com/btcsuite/btcd/blob/ec401d00a149d817e227b6afbc35b2399aca489d/txscript/standard.go#L991-L1005

Reply:
https://github.com/btcsuite/btcd/blob/ec401d00a149d817e227b6afbc35b2399aca489d/txscript/standard.go#L222-L235